test/cpp/jit/test_file_format.cpp

```diff
@@ -0,0 +1,124 @@
+#include <torch/csrc/jit/mobile/file_format.h>
+
+#include <gtest/gtest.h>
+
+#include <sstream>
+
+// Tests go in torch::jit
+namespace torch {
+namespace jit {
+
+TEST(FileFormatTest, IdentifiesFlatbufferStream) {
+  // Create data whose initial bytes look like a Flatbuffer stream.
+  std::stringstream data;
+  data << "abcd" // First four bytes don't matter.
+       << "PTMF" // Magic string.
```

Comment:
Interesting that flatbuffers uses bytes 4-7 while zip uses 0-3 as the magic string.

Reply:
It is a little weird, but I could see why they did it. By putting the offset to the data in the first word, files could contain an arbitrarily-large header starting with the magic, and the flatbuffer parsing logic wouldn't need to change.